packages/service/src/ui-service/storage/storage-account-service.ts

```diff
@@ -0,0 +1,49 @@
+import { ApiVersion, Endpoints } from "../constants";
+import { AbstractHttpService } from "../http-service";
+import { ResourceListResponse } from "../view/resource-models";
+import { StorageAccount } from "./storage-account-models";
+
+export interface StorageAccountService {
+    list(subscriptionId: string): Promise<StorageAccount[]>;
+    get(id: string): Promise<StorageAccount | null>;
+    create(account: StorageAccount): Promise<void>;
+    remove(account: StorageAccount): Promise<void>;
+    update(account: StorageAccount): Promise<void>;
+}
+
+export class StorageAccountServiceImpl
+    extends AbstractHttpService
+    implements StorageAccountService
+{
+    public async list(subscriptionId: string): Promise<StorageAccount[]> {
+        const response = await this.httpClient.get(
+            `${Endpoints.arm}/subscriptions/${formatSubscriptionId(
```

Comment:
The call to formatSubscriptionId seems out of place. This function should only accept a valid subscription ID.